### PR TITLE
Code cleanup: Remove references to FxCop

### PR DESCRIPTION
**Pull request by ladipro**
### Context

FxCop is deprecated and many of the constructs in the code base that claim to have been done to "make FxCop happy" are not needed anymore.

### Changes Made

- Removed FxCop exclusion files.
- Removed random references to FxCop, primarily from unit tests.
- Removed FxCop from comments and fixed the code as appropriate.

### Testing

CI and experimental insertion with the exclusion files removed.

### CODE REVIEW

File: src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj [hunk 1]
```diff
@@ -15,7 +15,6 @@
     <Compile Include="AssemblyInfo.cs" />
     <Compile Include="AssemblyResources.cs" />
     <Compile Include="Constants.cs" />
-    <Compile Include="FxCopExclusions\Microsoft.Build.Conversion.Suppressions.cs" />
```
Comment (rainersigwald) [inline src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:18]: Do you think it's worth making this change in `Deprecated/`? I usually try to leave that alone. On the other hand you already did it and it should be fine . . .

**Review by rainersigwald — APPROVED** [review]

File: src/Build/BackEnd/Components/Logging/EventSourceSink.cs [hunk 2]
```diff
@@ -217,66 +217,63 @@ public void Consume(BuildEventArgs buildEvent, int sinkId)
         /// </summary>
         public void Consume(BuildEventArgs buildEvent)
         {
-            // FXCop may complain that there are unecessary casts here, and there are, but
-            // using "as" and allocating another variable for each event is extremely costly
-            // and is much slower then this approach even with the additional casts
-            if (buildEvent is BuildMessageEventArgs)
+            if (buildEvent is BuildMessageEventArgs buildMessageEvent)
```
Comment (Forgind) [inline src/Build/BackEnd/Components/Logging/EventSourceSink.cs:220]: Nit:
Maybe nicer as a switch statement?

**Review by Forgind — APPROVED** [review]

File: src/Build/BackEnd/Components/Logging/EventSourceSink.cs (hunk 2 above)
Comment (ladipro) [reply src/Build/BackEnd/Components/Logging/EventSourceSink.cs]: I completely forgot about this feature. Done, thank you! Too bad switch expressions cannot be used here, they would make it even nicer.

File: src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj (hunk 1 above)
Comment (ladipro) [reply src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj:18]: I would lean towards making the change. Although not under active development the code still benefits from cleanup (a bit faster git operations, a bit faster build).

File: src/Deprecated/Engine/Engine/EventSource.cs [hunk 3]
```diff
@@ -520,40 +520,57 @@ internal void RaiseAnyEvent(object sender, BuildEventArgs e)
         }
 
         /// <summary>
-        /// Raises the given event to all registered loggers. This method up-cast the events
+        /// Raises the given event to all registered loggers. This method down-casts the events
```
Comment (rainersigwald) [inline src/Deprecated/Engine/Engine/EventSource.cs:523]: Wait, isn't this an upcast? to the more specific type? That matches my understanding of the "direction".

Comment (Forgind) [reply src/Deprecated/Engine/Engine/EventSource.cs:523]: [StackOverflow](https://stackoverflow.com/questions/23414090/what-is-the-difference-between-up-casting-and-down-casting-with-respect-to-class) says upcasting is to a supertype. I wasn't familiar with either terminology; maybe just change it to "This method casts all the events extracted from the queue to their more specific form" or something like that?

Comment (rainersigwald) [reply src/Deprecated/Engine/Engine/EventSource.cs:523]: huh, it appears to be used that way in other dotnet repos too. Yet more fuel for my "don't use terminology that implies a direction" fire.

Comment (ladipro) [reply src/Deprecated/Engine/Engine/EventSource.cs]: I've replaced it with direction-free wording.

(Reminds me of my confusion every time someone talks about LEFT JOIN and RIGHT JOIN.)

**Pull request merged**